react/ProductSearchContextProvider.js

```diff
@@ -172,10 +179,20 @@ class ProductSearchContextProvider extends Component {
         variables={queryField ? customSearch : defaultSearch}
         notifyOnNetworkStatusChange
       >
-        {searchQueryProps => {
-          const { data, loading } = searchQueryProps
+        {searchQuery => {
+          const { data, loading } = searchQuery
           const { search } = data || {}
           const { titleTag, metaTagDescription } = search || {}
+
+          // quick fix, needs further investigation
+          if (!loading && isEmpty(data) && this.state.retries < MAX_QUERY_RETRIES) {
```

Comment:
I don't like this approach =/ You need to try another approach. Tell me what you need to investigate this.